### PR TITLE
fix(dracut-systemd): include systemd-cryptsetup module when needed

### LINKED ISSUE #563: LVM + LUKS cryptsetup broken for dracut-ng-102 and 103
I have physical and virtual hosts using LVM with LUKS cryptsetup setup:
 1. LUKS / cryptsetup partition
 2. LVM based root and swap volumes on above LUKS partition

Gentoo Linux as distro with systemd as init-system and systemd-boot as boot-manager.

After migrating to dracut-ng-102 (from dracut-ng-60) all these systems refuse to boot. On the KVM/Qemu VM hosts i am able to capture the following error:

[ 1.239849] dracut-initqueue[418]: Failed to start systemd-cryptsetup@luks\x2dcde09b75\x2dd8f7\x2d4be0\x2d88ff\x2deb9c7f36327f.service: Unit systemd-cryptsetup@luks\x2dcde09b75\x2dd8f7\x2d4be0\x2d88ff\x2deb9c7f36327f.service not found.

I can reproduce the error with dracut-ng-102 and 103.
Dracut-ng-101 works as expected.

All the above with unchanged config-files.

The above error appears regardless if use dracut to generate a standard initrd-file or an signed UKI for direct UEFI-secure-boot.

The attached configs and logs are from my laptop which makes use of  the signed UKI and UEFI-secure-boot.

[blkid.txt](https://github.com/user-attachments/files/16491150/blkid.txt)
[dracut.conf.txt](https://github.com/user-attachments/files/16491151/dracut.conf.txt)
[install.conf.txt](https://github.com/user-attachments/files/16491152/install.conf.txt)
[lsblk-f.txt](https://github.com/user-attachments/files/16491153/lsblk-f.txt)
[make-install-dracut-101.txt](https://github.com/user-attachments/files/16491154/make-install-dracut-101.txt)
[make-install-dracut-102.txt](https://github.com/user-attachments/files/16491155/make-install-dracut-102.txt)

Maybe this issue is related to [issue](https://github.com/dracut-ng/dracut-ng/issues/451).

### PR DESCRIPTION
**Pull request by jozzsi**
Make `systemd-cryptsetup` a dependency of the `crypt` module when systemd dracut module is included.
Improve tests with crypt dracut module to work in hostonly mode. Enable running these test both in hostonly and non-hostonly mode.

## Checklist
- [x] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #563

### CODE REVIEW

**Comment by LaszloGombos** [comment]
Thanks @jozzsi . I would prefer a solution similar to what was proposed at https://github.com/dracut-ng/dracut-ng/issues/563#issuecomment-2268780785 to stay consistent on how module dependencies are expressed.

**Review by LaszloGombos — APPROVED** [review]
This approach makes sense to me. Thanks @jozzsi to fix the tests - had we had these tests in this form at the time the change was made, these tests would have at least signal the regression.

@floppym @ferringb @stenker - thanks for raising this issue. I saw your discussion at https://bugs.gentoo.org/937326 .  Do you have any additional feedback or concern before landing this PR ? 

Other distributions run into issues related this change as well - see https://github.com/microsoft/azurelinux/pull/10103

**Review by Conan-Kudo — APPROVED** [review]
This seems reasonable to me as well. Absent any reason from @dracut-ng/gentoo-maint in the next day to not merge this (as they were specifically called out earlier), I'll merge this tomorrow.

**Comment by ferringb** [comment]
@LaszloGombos if y'all are fine w/ the mixed deps (forward and reverse), no complaints from me.  I don't mean offense, but the modules inclusion system is kind of a pain in the ass to trace at this point- I'm not sure adding a forward dep would make it worse although I think keeping it to reverse will make the inevitable refactoring easier.

**Review by AndrewAmmerlaan — APPROVED** [review]
This makes sense to me

**Pull request merged**
